src/LPM.py

```diff
@@ -63,22 +65,21 @@ def colored(text, color):
         def cprint(text, color):
             print(text)
 
-    # Prepend the @loupeteam prefix to all package names, and split any @version suffixes off into separate struct.
-    packages = []
-    packageVersions = []
+    # Build array of PackageNameVersion objects from args
+    packageNameObjectList = []
     if(args.packages):   
         for item in args.packages:
-            # Force package name to lowercase, by convention
-            item = item.lower()
-            # If the @ character is present, it means there's a version specifier.
-            if('@' in item):
-                splitItem = item.split('@')
-                packages.append('@loupeteam/' + splitItem[0])
-                packageVersions.append(splitItem[1])
-            # Othersie, version string is empty.
-            else:
-                packages.append('@loupeteam/' + item)
-                packageVersions.append('')
+            try:
+                packageNameObjectList.append(Package.PackageNameVersion(item))
+            except:
+                print(colored(f'Could not interpret \"{item}\" as a valid package', 'red'))
+        
+        # Return if no valid packages
+        if len(packageNameObjectList) == 0:
+            return
+    packagesFullNames =  [p.fullName    for p in packageNameObjectList]  
+    packagesBaseNames =  [p.baseName    for p in packageNameObjectList]  
+    packagesVersions  =  [p.version     for p in packageNameObjectList]    
```

Comment:
I don't get why have you implemented `PackageNameVersion` class to have all info regarding a package in one place but then decided to split a list of package into three different lists?